### ISSUE TITLE
Feature request: Allow Post body to be a target parameter

### ISSUE THREAD

**Issue by danielmotaleite** [prometheus/blackbox_exporter#380]
Right now to test several http POST we need to configure several modules, each with their own `body` payload.

So we need to test several POST requests and seems a waste to have exactly the same thing multiple times, except a small `body` change

It would be great that the body could the a target parameter and reuse the http_post for all targets

**Comment by brian-brazil** [comment]
Thanks for your suggestion. It's by design that the config file is the only way to do this, as the blackbox exporter is meant to test many targets with the same module and it should not be an open proxy.

**Comment by danielmotaleite** [comment]
how about one (or several) variable that could be replaced in the post?
one could define some variable and value (eg: healthcheck: qwerty ) and it could be replaced in the `body` (eg: body: '{"healthcheck":"{{ company }}" }' )

This would allow reusing modules and still only allow a restricted POST test

**Comment by danielmotaleite** [comment]
True, but any change requires change in 2 files, one in blackbox and another in prometheus... this kind of changes allow one to only change one service and keep the changing parts in one place. Blackbox would keep the generic module rules. Add docker to the mix and you have to deploy config, rebuild docker and deploy it... just to add/change a new service.

Of course we can survive this, but IMHO it would be easier to push at least some parts of this to the target

**Comment by brian-brazil** [comment]
If you've got bespoke health checks for every single service, then that's one of the things that you have to deal with.

**Comment by vgallissot** [comment]
Hi @brian-brazil,

Given the fact that, by design, blackbox exporter won't act as a proxy - as you said above - what exporter can we use to do functional tests ?

I.E:
* I have an HTTP API which must return a specific response on a specific route
* I want to test and monitor this API's route, with also response time and all features blackbox exporter provides

Thanks for your quick response,
Vincent

**Comment by brian-brazil** [comment]
If the input is static, you can have it as module configuration in the blackbox exporter. If you want to do something fancier than that you might want to look at the webdriver exporter, or write your own.

**Comment by vgallissot** [comment]
OK, thanks

**Comment by xujl1062** [comment]
I have the same request，and added the following sections to the main.go(138) 
```
body := params.Get("body")
if body != "" {
    module.HTTP.Body = body
}
```
then, We can fetch post body from the prometheus sd_config by relabel func
